src/main/scala/edu/ie3/simona/agent/participant/evcs/EvcsAgentFundamentals.scala

```diff
@@ -357,42 +353,188 @@ protected trait EvcsAgentFundamentals
         lastEvs
       )
 
-    val (power, evModelsCalculated) =
+    val (result, evModelsCalculated) =
       evcsModel.calculatePowerAndEvSoc(
-        currentTick,
+        tick,
         voltage,
         relevantData
       )
 
-    {
-      val departedEvs = calculateDepartedEvs(
-        evModelsCalculated,
-        evcsMovementsData.movements
-      )
-      if (departedEvs.nonEmpty) {
-        evServiceRef ! DepartedEvsResponse(
-          evcsModel.uuid,
-          departedEvs
-        )
+    val (departingEvs, stayingEvs) =
+      evModelsCalculated.partition { ev =>
+        // EV has been parked up until now and has now departed
```

Comment:
One could argue about semantics here since MobSim determined the departing EVs and that could justify the use of past tense here. But in the end I agree your suggestion seems sensible within the temporal framework that we use here.